### PR TITLE
Bump cpi to 1.22.5

### PR DESCRIPTION
**Pull request by lubronzhan**
## What this PR does / why we need it

Bump CPI to 1.22.5 
Ability to exclude subnets for internal/external node ip selection 

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Bump CPI to 1.22.5, add ability to exclude subnets for internal/external node ip selection 
```

## Which issue(s) this PR fixes

Fixes: #

### CODE REVIEW

**Comment by lubronzhan** [comment]
Hi @seemiller could you help take a look?

**Comment by jpmcb** [comment]
Since we added @HanFa to the team of @vmware-tanzu/pkg-vsphere-cpi-owners, is it possible for them to approve? Nick and I are only apart of those teams currently for administrative purposes.

**Comment by lubronzhan** [comment]
Yeah sure! @HanFa  Could you review this? cc @tylerschultz

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/test/go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/vmware-tanzu/community-edition/addons/packages/vsphere-cpi/1.22.4/test
```
Comment (HanFa) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/test/go.mod:1]: we don't need to update the package name for tests in `1.23.0-alpha.1`

**Review by HanFa — DISMISSED** [review]
LGTM thanks @lubronzhan

**Comment by lubronzhan** [comment]
Hi @jpmcb We still need a TCE owner approval to merge the PR. @HanFa already approved. Could you also approve? Thanks

**Comment by jpmcb** [comment]
I'm a bit confused why v1.22.4 was removed? Wouldn't we want to keep that version in source control and introduce a new directory with the new version v1.22.5?

**Comment by lubronzhan** [comment]
I saw the previous pattern is to keep the newest patch, so I thought that was the pattern, so that customer should only use the newest with the bug fix. For source control point of view, then yes we should keep all the version. Let me update this

**Comment by lubronzhan** [comment]
Hi @jpmcb could you help take a look again?

File: addons/packages/vsphere-cpi/1.22.5/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+|-------|-------------------|-------------|
+| `vsphereCPI.server` | Required | The IP address or FQDN of the vSphere endpoint. Default value is `null`. |
+| `vsphereCPI.datacenter` | Required | The datacenter in which VMs are created/located. Default value is `null`. |
+| `vsphereCPI.username` | Required | Username used to access a vSphere endpoint. Default value is `null`. |
+| `vsphereCPI.password` | Required | Password used to access a vSphere endpoint. Default value is `null`. |
+| `vsphereCPI.tlsThumbprint` | Optional | The cryptographic thumbprint of the vSphere endpoint's certificate. Default value is `""`. |
+| `vsphereCPI.insecureFlag` | Optional | The flag that disables TLS peer verification. Default value is `False`. |
+| `vsphereCPI.ipFamily` | Optional | IP family. Default value is `null`. |
+| `vsphereCPI.vmInternalNetwork` | Optional | Internal VM network name. Default value is `null`. |
+| `vsphereCPI.vmExternalNetwork` | Optional | External VM network name. Default value is `null`. |
```
Comment (adobley) [inline addons/packages/vsphere-cpi/1.22.5/README.md:27]: We should add the new configurable values to this list:
```
| `vsphereCPI.vmExcludeInternalNetworkSubnetCidr` | Optional | Comma separated list of internal network subnets to exclude from node IP selection. Default value is `null`. |
| `vsphereCPI.vmExcludeExternalNetworkSubnetCidr` | Optional | Comma separated list of external network subnets to exclude from node IP selection. Default value is `null`. |
```

File: addons/packages/vsphere-cpi/1.22.5/bundle/config/values.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  server: null
+  datacenter: null
+  username: null
+  password: null
+  region: null
+  zone: null
+  insecureFlag: False
+  ipFamily: null
+  vmInternalNetwork: null
+  vmExternalNetwork: null
```
Comment (adobley) [inline addons/packages/vsphere-cpi/1.22.5/bundle/config/values.yaml:16]: I believe we need to add the defaults for the new exclude values here:
```
  vmExcludeInternalNetworkSubnetCidr: null
  vmExcludeExternalNetworkSubnetCidr: null
```

File: addons/packages/vsphere-cpi/1.22.5/bundle/config/vsphereconf.lib.txt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+(@ if values.vsphereCPI.region or values.vsphereCPI.zone: -@)
+[Labels]
+(@ if values.vsphereCPI.region: -@)
+region = "(@=values.vsphereCPI.region @)"
+(@ end -@)
+(@ if values.vsphereCPI.zone: -@)
+zone = "(@=values.vsphereCPI.zone @)"
+(@ end -@)
+(@ end -@)
+(@ if values.vsphereCPI.vmInternalNetwork or values.vsphereCPI.vmExternalNetwork: -@)
```
Comment (adobley) [inline addons/packages/vsphere-cpi/1.22.5/bundle/config/vsphereconf.lib.txt:30]: Line 30-38 needs to be updated to pass the new configuration values through:
```
(@ if values.vsphereCPI.vmInternalNetwork or values.vsphereCPI.vmExternalNetwork or values.vsphereCPI.vmExcludeInternalNetworkSubnetCidr or values.vsphereCPI.vmExcludeExternalNetworkSubnetCidr: -@)
[Nodes]
(@ if values.vsphereCPI.vmInternalNetwork: -@)
internal-vm-network-name = "(@=values.vsphereCPI.vmInternalNetwork @)"
(@ end -@)
(@ if values.vsphereCPI.vmExternalNetwork: -@)
external-vm-network-name = "(@=values.vsphereCPI.vmExternalNetwork @)"
(@ end -@)
(@ if values.vsphereCPI.vmExcludeInternalNetworkSubnetCidr: -@)
exclude-internal-network-subnet-cidr = "(@=values.vsphereCPI.vmExcludeInternalNetworkSubnetCidr @)"
(@ end -@)
(@ if values.vsphereCPI.vmExcludeExternalNetworkSubnetCidr: -@)
exclude-external-network-subnet-cidr = "(@=values.vsphereCPI.vmExcludeExternalNetworkSubnetCidr @)"
(@ end -@)
(@ end -@)
```

File: addons/packages/vsphere-cpi/1.22.5/test/vsphere_cpi_test.go [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv5`
+			})
+
+			It("errors when rendering", func() {
+				Expect(yttRenderErr).To(MatchError(ContainSubstring("vsphereCPI ipFamily should be one of \"ipv4\", \"ipv6\", \"ipv4,ipv6\", or \"ipv6,ipv4\" if provided")))
+			})
+		})
```
Comment (adobley) [inline addons/packages/vsphere-cpi/1.22.5/test/vsphere_cpi_test.go:231]: We should be able to add the Node IP Selectors tests from 1.23.0 to cover the new exclude behavior https://github.com/vmware-tanzu/community-edition/blob/5aaf4df530fce01545f823492e760229bbbbf644/addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go#L348-L471

**Comment by lubronzhan** [comment]
Thanks @adobley. I made those change as well!

File: addons/packages/vsphere-cpi/1.22.5/bundle/config/vsphereconf.lib.txt [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+(@ end -@)
+(@ end -@)
+(@ if values.vsphereCPI.vmInternalNetwork or values.vsphereCPI.vmExternalNetwork or values.vsphereCPI.vmExcludeInternalNetworkSubnetCidr or values.vsphereCPI.vmExcludeExternalNetworkSubnetCidr: -@)
+[Nodes]
+(@ if values.vsphereCPI.vmInternalNetwork: -@)
+internal-vm-network-name = "(@=values.vsphereCPI.vmInternalNetwork @)"
+(@ end -@)
+(@ if values.vsphereCPI.vmExternalNetwork: -@)
+external-vm-network-name = "(@=values.vsphereCPI.vmExternalNetwork @)"
+(@ end -@)
```
Comment (adobley) [inline addons/packages/vsphere-cpi/1.22.5/bundle/config/vsphereconf.lib.txt:37]: We need to add these INI lines for the exclude config here too https://github.com/vmware-tanzu/community-edition/blob/5aaf4df530fce01545f823492e760229bbbbf644/addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/vsphereconf.lib.txt#L38-L43

**Review by adobley — DISMISSED** [review]
Other than that last INI bit, LGTM

**Comment by lubronzhan** [comment]
Good catch. Sorry

**Comment by lubronzhan** [comment]
Hi @jpmcb  Could you help take a look again?
Thank you!

**Comment by tylerschultz** [comment]
LGTM - thanks for everyone's help!

File: .github/dependabot.yml [hunk 7]
```diff
@@ -182,7 +182,7 @@ updates:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
-    directory: "addons/packages/vsphere-cpi/1.22.4/test"
+    directory: "addons/packages/vsphere-cpi/1.22.5/test"
```
Comment (jpmcb) [inline .github/dependabot.yml:185]: I think we actually want to keep this line as is and add _another_ chunk that corrisponds to the new package being added. We still want to have the existing chunk for v1.22.4 _in addition_ to the one for v1.22.5

**Review by jpmcb — DISMISSED** [review]
Looks good to me! Small nit in the dependabot yamls

File: .github/dependabot.yml (hunk 7 above)
Comment (lubronzhan) [reply .github/dependabot.yml:185]: Oh, what's this file used for?
Let me update it

Comment (jpmcb) [reply .github/dependabot.yml:192]: It's used to configure GitHub's dependabot, an automated dependency scanning / updating system. These configs ignore tanzu-framework which the TCE team has been bumping manually

**Review by jpmcb — COMMENTED** [review]
LGTM - can you squash your commits and then we'll be good to merge!

**Comment by mcwumbly** [comment]
@jpmcb process question: wouldn't "squash and merge" do the trick there? 
https://tanzucommunityedition.io/docs/latest/contribute/contributing/#merging-commits

is it preferred that contributors do the squashing themselves first for some reason? (and if so, should we capture that in those contributing docs)?

**Comment by lubronzhan** [comment]
+1 to what David said, I was gonna use that button. 
Anyway. I squashed my commits. Thanks @jpmcb

**Comment by jpmcb** [comment]
> is it preferred that contributors do the squashing themselves first for some reason?

Good question! We do have this chunk i[n our docs](https://tanzucommunityedition.io/docs/latest/contribute/contributing/): 

>Maintainers should prefer to merge pull requests with the Squash and merge option. This option is preferred for a number of reasons. First, it causes GitHub to insert the pull request number in the commit subject which makes it easier to track which PR changes landed in. Second, it gives maintainers an opportunity to edit the commit message to conform to Tanzu Community Edition standards and general good practice. Finally, a one-to-one correspondence between pull requests and commits makes it easier to manage reverting changes and increases the reliability of bisecting the tree (since CI runs at a pull request granularity).

But it's not a strict requirement that the contributor squash their own commits. Admittedly, it's my personal preference since I can't always anticipate and write the best commit message when squashing a number of commits using the "Squash and merge" button. For me, it's almost always easier to merge something that is one commit already with a single cohesive commit message. But that's more philosophical than anything. Feel free to ask us to just "squash and merge" when needed

**Review by jpmcb — APPROVED** [review]

**Pull request merged**